### PR TITLE
Changes to subject indicators

### PR DESCRIPTION
**Pull request by SandGrainOne**
## Description
Making multiple changes to how we handle subject prefixes and values:
- CreatedBy and Consumer
    - Added support for `/organisation/` in access subject authorization. 
    - Storing new subscriptions for an organisation as `/organisation/` instead of `/party/`. One thing to note is that there are no subscription in any environment where CreatedBy or Consumer data have the `/party/` prefix. They are all either `/org/` or `/user/`. This means that at no point in time has anyone successfully created an organisation subscription.
- Alternative subject
    - Added support for alternative subject using `/organisation/` as prefix. Keeping `/org/` for backwards compatibility. This might be in use by external parties when setting subject filter on a subscription.

Other changes:
- `SubscriptionService.GetAllSubscriptions()` where the only method taking consumer as input. Changed to use `IClaimPrincipalProvider` instead.
- Removed support for Hungarian prefixes on variable names

## Related Issue(s)
- #469 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green
- [ ] Manual testing done (required)

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Events/Authorization/XacmlMapperHelper.cs [hunk 1]
```diff
@@ -17,15 +17,17 @@ public static class XacmlMapperHelper
     private const string UserPrefix = "/user/";
     private const string OrgPrefix = "/org/";
     private const string PartyPrefix = "/party/";
-    private const string OrganizationPrefix = "/organization/";
+    private const string OrganisationPrefix = "/organisation/";
```
Comment (acn-sbuad) [inline src/Events/Authorization/XacmlMapperHelper.cs:20]: shouldn't this also be 'z'  ? ref comment on line 27.

Comment (SandGrainOne) [reply src/Events/Authorization/XacmlMapperHelper.cs:20]: The comment is for a specific value used by the Authorization logic. I would have used `s` if I could.
I'll try to make it more obvious what I ment.

File: src/Events/Services/AppSubscriptionService.cs [hunk 2]
```diff
@@ -137,16 +133,26 @@ private async Task<string> GetPartyFromAlternativeSubject(string alternativeSubj
         {
             int partyId = 0;
 
-            if (alternativeSubject.StartsWith(OrganisationPrefix))
+            /* Both OrgPrefix and OrganisationPrefix is assumed to give us the organisation number of an 
+             * organisation. OrgPrefix is NOT for the application owner acronym like it is many other 
+             * places. OrgPrefix was kept as is when adding OrganisationPrefix for backwards compability.
+             */
+
+            if (alternativeSubject.StartsWith(OrgPrefix))
```
Comment (acn-sbuad) [inline src/Events/Services/AppSubscriptionService.cs:141]: Can we consider handling the orgPrefix and OrganisationPrefix in the same if-clause to better demonstrate that they represent the same "thing" ?

Comment (acn-sbuad) [reply src/Events/Services/AppSubscriptionService.cs:141]: Alternatively, we could have the controller be responsible for mapping the alternative subject to a format that the service can handle. E.g. mapping all /org/ to /organisation/. 
Limiting the backward compatibility to the controller and making it clear there that use of /org/ is deprecated?

Comment (SandGrainOne) [reply src/Events/Services/AppSubscriptionService.cs:143]: I like the code flow the way it is, but I moved the to cases closer together and improved the explanation.

**Review by acn-sbuad — APPROVED** [review]
🎈

**Pull request merged**